### DISCUSSION TITLE
Add OIDC to existing RKE2 cluster.

### DISCUSSION THREAD

**Discussion by easpeagle** [rancher/rke2#3861] [Q&A] [answered]
Hi Team,

I'm trying to reason out how I would add oidc support the kube-apiserver in my cluster. I tried updating /var/lib/rancher/rke2/agent/pod-manifests/kube-apiserver.yaml ... but that just seem to be wiped out at server start. So, I'm guess that is recreated every time.

What's the correct method for making this happen?

Thanks!

**Comment by brandond** [comment, marked as answer]
If you need to add custom flags to the apiserver, you should use the `--kube-apiserver-arg` flag or `kube-apiserver-arg: []` config file key. The static pod manifests are all managed by RKE2 and should not be manually edited.

**Reply by easpeagle** [reply]
That was helpful... thanks!
